src/engine/effects/engineeffect.h

```diff
@@ -1,5 +1,6 @@
 #pragma once
 
+#include <QDialog>
```

Comment:
forward decl suffices actually

Reply:
So just `class QDialog;`? Are forward decls generally preferred for headers?

Reply:
its a tradeoff IMO. If the header is big (usually the case with Qt headers) and the header where its included is used in lots of other headers and you only heap-allocate the class anyways (with a unique_ptr for example), then I'd forward-declare them. In this case it would make sense because you really only need to name it in order to return from a single member function. So pulling in an entire header does come with quite some bloat IMO.